sdks/python/apache_beam/runners/worker/log_handler.py

```diff
@@ -106,7 +120,19 @@ def _write_log_entries(self):
       if log_entries:
         yield beam_fn_api_pb2.LogEntry.List(log_entries=log_entries)
 
-  def _read_log_control_messages(self, log_control_iterator):
-    # TODO(vikasrk): Handle control messages.
-    for _ in log_control_iterator:
-      pass
+  def _read_log_control_messages(self):
+    while True:
+      log_control_iterator = self.connect()
+      if self._dropped_logs > 0:
+        logging.warn("Dropped %d logs while logging client disconnected",
```

Comment:
I do not want to block this PR. Let's merge. but I have a follow up question.

If the queue is already full, would not this message be also dropped as well. (Unless the log write gets scheduled in between connection and this log message?

Reply:
It's not guaranteed to make it, although in my tests it consistently does. Once we call connect() we start to drain from the queue. So so long as something has drained by the point we get to the warn it will show up.

Alternatively, we could write to stderr as we do to report the connection error as that should be reliable (but might not be as accessible as real logs). Happy to hear other suggestions as well.